### PR TITLE
fix npe related to missing configuration sections

### PR DESCRIPTION
**Pull request by HgeX**
Hey! 👋 I ran into an issue after updating AACH from 6.7.2 to 7.0.0 (7.0.1 produced the same error).

```[08:52:21] [Server thread/ERROR]: Error occurred while enabling AdvancedAchievements v7.0.1 (Is it up to date?)
java.lang.NullPointerException: null
	at com.hm.achievement.config.ConfigurationParser.parseAchievements(ConfigurationParser.java:289) ~[?:?]
	at com.hm.achievement.config.ConfigurationParser.loadAndParseConfiguration(ConfigurationParser.java:90) ~[?:?]
	at com.hm.achievement.lifecycle.PluginLoader.loadAdvancedAchievements(PluginLoader.java:142) ~[?:?]
	at com.hm.achievement.AdvancedAchievements.onEnable(AdvancedAchievements.java:49) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Paper-506]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Paper-506]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Paper-506]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Paper-506]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Paper-506]
	at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:466) ~[patched_1.16.5.jar:git-Paper-506]
	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:239) ~[patched_1.16.5.jar:git-Paper-506]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:941) ~[patched_1.16.5.jar:git-Paper-506]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:175) ~[patched_1.16.5.jar:git-Paper-506]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

After forking the project and adding debug messages, it turned out that this is related to missing configuration sections. Maybe `YamlConfiguration#getConfigurationSection` behaves differently compared to older spigot/paper versions, but I haven't touched the config in a while, so this error came to me as a not so pleasent surprise. 

The problem is that every achievement category has been disabled like this:
```yaml
Category: []
```
I guess this isn't the proper way of doing it, but I really don't want to reconfigure the whole thing... Anyways, my solution to this is to replace `getConfigurationSection("key").getKeys(false).isEmpty()` to `isConfigurationSection("key")`. It should do the same thing but without producing errors. I have tested it on paper 1.16.5 and on craftbukkit 1.7.2 to see if this function even exists on such an old version, but I haven't encountered any issues so far. *(Altough I discovered something else on 1.7.2, I'll open a PR for that as well, if you don't mind :) )*

### CODE REVIEW

**Review by PyvesB — CHANGES_REQUESTED** [review]
Hello @HgeX! :wave:

Thanks for your interest in the project!

There are two way of disabling a category in Advanced Achievements:
* either by adding it to the `DisabledCategories` list.
* either by making it an empty YAML configuration section, e.g. `Category: {}`.

Unfortunately, using `isConfigurationSection` as the main guard as in your pull request breaks the second bullet point, empty categories are no longer disabled.

Using `Category: []` means empty list in YAML, which is technically a user mistake as achievement categories are not lists. If you really want to deal with this, you would need to combine both `isConfigurationSection` with `isEmpty`. 😉

**Comment by HgeX** [comment]
Hey! Thank you for your reply :)

> Using Category: [] means empty list in YAML

I really should drink more coffee these days...

Anyways, I'd happily implement your suggestion, if you don't mind of course. *(Maybe even add a warning, if a config section is missing? Because right now it results in an NPE, which is not necessarily ideal. What do you think about this?)*

Have a nice day! :)

**Comment by PyvesB** [comment]
> Maybe even add a warning, if a config section is missing? Because right now it results in an NPE, which is not necessarily ideal.

I would tend to say that if the user specified `Category: []`, there's little doubt about the fact that they actually wanted to disable the category. The plugin should just do that for them transparently, I don't think there's much benefit in adding a warning. :)

**Comment by HgeX** [comment]
Yeah, you're right... It's a bad idea. I'll combine those two checks together and push it tomorrow. :)

**Comment by HgeX** [comment]
Hey! 👋

Sorry for the delay, I was busy yesterday! :( Anyway, the two checks are now combined and it seems to be working on my server.

I'm terrible at naming things, so if you can come up with a better name for that function please let me know! ;)

File: advanced-achievements-plugin/src/main/java/com/hm/achievement/config/ConfigurationParser.java [hunk 1]
```diff
@@ -299,11 +311,12 @@ private void parseAchievements() throws PluginLoadError {
...
 		for (MultipleAchievements category : MultipleAchievements.values()) {
 			if (!disabledCategories.contains(category)) {
-				Set<String> subcategories = mainConfig.getConfigurationSection(category.toString()).getKeys(false);
-				if (subcategories.isEmpty()) {
+				if (getSectionKeys(category.toString()).isEmpty()) {
 					disabledCategories.add(category);
 					continue;
 				}
+
+				Set<String> subcategories = mainConfig.getConfigurationSection(category.toString()).getKeys(false);
```
Comment (PyvesB) [inline advanced-achievements-plugin/src/main/java/com/hm/achievement/config/ConfigurationParser.java:319]: As we're calling `getSectionKeys` in the above conditional, could we not extract that to a local variable and reuse it here?

**Comment by PyvesB** [comment]
No worries for the delay, there is no emergency here! `getSectionKeys` sounds like a perfectly sensible name to me. 😉

File: advanced-achievements-plugin/src/main/java/com/hm/achievement/config/ConfigurationParser.java (hunk 1 above)
Comment (HgeX) [reply advanced-achievements-plugin/src/main/java/com/hm/achievement/config/ConfigurationParser.java:319]: Oops, I missed that!

**Comment by HgeX** [comment]
Okay, I think it's done. :)

**Review by PyvesB — APPROVED** [review]
Looks good to me, thank you!

I'm curious to see what other problems you've spotted on Craftbukkit 1.7.2. Years and years ago I marked the plugin as being compatible with version 1.7.9 as a minimum, but I don't remember what drove that decision at the time. A useful exercise could be to remove the Spigot dependency from the Maven dependencies of Advanced Achievements and instead try compiling with the 1.7.2 JAR file - that would highlight some potential problems. 😉

**Comment by HgeX** [comment]
Thank you for merging this PR :) 

> A useful exercise could be to remove the Spigot dependency from the Maven dependencies of Advanced Achievements and instead try compiling with the 1.7.2 JAR file - that would highlight some potential problems. 😉

I can do that if you'd like. On 1.7.2 Bukkit#getOnlinePlayers returns Player[] instead of Collection<? extends Player>, that leads to a couple of `NoSuchMethodErrors` here and there, but the issue doesn't persist on 1.7.10, so I figured this isn't really huge problem.

**Comment by PyvesB** [comment]
> On 1.7.2 Bukkit#getOnlinePlayers returns Player[] instead of Collection<? extends Player>, that leads to a couple of `NoSuchMethodErrors` here and there, but the issue doesn't persist on 1.7.10, so I figured this isn't really huge problem.

That will be a bit annoying to work around. Do you actually use 1.7.2 on a real server? If not, I suggest we don't bother adding compatibility for something we don't know anyone for whom it will be useful.

**Comment by HgeX** [comment]
Yeah I totally agree on that. My server is on 1.16.5, I installed AACH on that ancient server just for the sake of testing.

**Comment by PyvesB** [comment]
Okay, let's leave that for now then. Do feel free to contribute other stuff in the future. 😉

**Pull request merged**